pkg/test/framework/components/istio/config.go

```diff
@@ -63,20 +63,21 @@ var (
 	helmValues string
 
 	settingsFromCommandline = &Config{
-		ChartRepo:          DefaultIstioChartRepo,
-		SystemNamespace:    DefaultSystemNamespace,
-		IstioNamespace:     DefaultSystemNamespace,
-		ConfigNamespace:    DefaultSystemNamespace,
-		TelemetryNamespace: DefaultSystemNamespace,
-		PolicyNamespace:    DefaultSystemNamespace,
-		IngressNamespace:   DefaultSystemNamespace,
-		EgressNamespace:    DefaultSystemNamespace,
-		DeployIstio:        true,
-		DeployTimeout:      0,
-		UndeployTimeout:    0,
-		ChartDir:           env.IstioChartDir,
-		CrdsFilesDir:       env.CrdsFilesDir,
-		ValuesFile:         E2EValuesFile,
+		ChartRepo:                 DefaultIstioChartRepo,
+		SystemNamespace:           DefaultSystemNamespace,
+		IstioNamespace:            DefaultSystemNamespace,
+		ConfigNamespace:           DefaultSystemNamespace,
+		TelemetryNamespace:        DefaultSystemNamespace,
+		PolicyNamespace:           DefaultSystemNamespace,
+		IngressNamespace:          DefaultSystemNamespace,
+		EgressNamespace:           DefaultSystemNamespace,
+		DeployIstio:               true,
+		DeployTimeout:             0,
+		UndeployTimeout:           0,
+		ChartDir:                  env.IstioChartDir,
+		CrdsFilesDir:              env.CrdsFilesDir,
+		ValuesFile:                E2EValuesFile,
+		EnableNamespacesByDefault: false,
```

Comment:
The name doesn't make it clear to me what is really going on. From the comments by @howardjohn it sounds like what we're really doing is selecting to use a custom injector, rather than the default. Would it make sense to rename this flag to something like `useCustomSidecarInjector`?

It appears that the custom injector is scoped to a namespace (e.g. istio-system). Is that right? If so, another way we could approach this is to make the variable be `--sidecarInjectorNamespace=<ns>`. This might be more flexible if we ever wanted to put the injector in its own namespace.

Reply:
I don't think `sidecarInjectorNamespace` makes sense

The new installer has two models:
1. inject everything that doesn't have `inject: false` or `istio-env: ...`
2. inject things with `istio-env: namespace` using the injector in namespace

So if you wanted mode 1 you would then pass `--sidecarInjectorNamespace=""`? but this is misleading, because the injector does exist, it just doesn't matter where the namespace is?

`useCustomSidecarInjector` seems good though

Reply:
Sure... `useCustomSidecarInjector` sgtm